### PR TITLE
Created auto assign logger for statistics

### PR DESCRIPTION
**Pull request by Kevma50287**
Resolves [APPEALS 38420](https://jira.devops.va.gov/browse/APPEALS-38420)

# Description
Add a CorrespondenceAutoAssignmentLogger class to keep track of counts and statistics.
Methods are stubbed and will be updated with the appropriate ActiveRecord objects and updates.

## Acceptance Criteria
- [x] Code compiles correctly

## Testing Plan

1. Go to [Jira Issue/Test Plan Link](https://jira.devops.va.gov/browse/JIRA-12345) or list them below

- [ ] For feature branches merging into master: Was this deployed to UAT?

# Best practices
## Code Documentation Updates
- [x] Add or update code comments at the top of the class, module, and/or component.

### Test Coverage
Did you include any test coverage for your code? Check below:
- [ ] RSpec
- [ ] Jest
- [ ] Other

### Code Climate
Your code does not add any new code climate offenses? If so why?
- [X] No new code climate issues added

### Monitoring
- [ ] Are performance metrics (e.g., response time, throughput) being tracked?
- [ ] Are key application components monitored (e.g., database, cache, queues)?
- [ ] Is there a system in place for setting up alerts based on performance thresholds?

### Logging
- [X] Are logs being produced at appropriate log levels (debug, info, warn, error, fatal)?
- [X] Are logs structured (e.g., using log tags) for easier querying and analysis?
- [X] Are sensitive data (e.g., passwords, tokens) redacted or omitted from logs?
- [X] Is log retention and rotation configured correctly?
- [X] Are logs being forwarded to a centralized logging system if needed?

### Auditing
- [ ] Are user actions being logged for audit purposes?
- [ ] Are changes to critical data being tracked ?
- [ ] Are logs being securely stored and protected from tampering or exposing protected data?

### Error Handling
- [ ] Are errors being caught and handled gracefully?
- [ ] Are appropriate error messages being displayed to users?
- [ ] Are critical errors being reported to an error tracking system (e.g., Sentry, ELK)?
- [ ] Are unhandled exceptions being caught at the application level ?

### Exception Handling
- [ ] Are custom exceptions defined and used where appropriate?
- [ ] Is exception handling consistent throughout the codebase?
- [ ] Are exceptions logged with relevant context and stack trace information?
- [ ] Are exceptions being grouped and categorized for easier analysis and resolution?

### CODE REVIEW

**Review by tradesmanhelix — COMMENTED** [review]
I think this is a good start and we can iterate from here 👍

File: app/services/correspondence_auto_assign_logger.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    @failed_attempts_count = 0
+    @failed_assignments = []
+    @successful_assignments = []
+  end
+
+  def begin_logging
+    create_batch_assignment
+  end
+
+  def end_logging
```
Comment (zurbergram) [inline app/services/correspondence_auto_assign_logger.rb:15]: Let's add a parameter here to indicate why this was called so then when you incorporate table usage it becomes the value that goes into the status column
```suggestion
  def end_logging(status)
```

File: app/services/correspondence_auto_assign_logger.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+class CorrespondenceAutoAssignLogger
+  def initialize
+    @total_attempts = 0
+    @failed_attempts_count = 0
+    @failed_assignments = []
+    @successful_assignments = []
+  end
+
+  def begin_logging
```
Comment (zurbergram) [inline app/services/correspondence_auto_assign_logger.rb:11]: Missing parameter of `user_id` to indicate who started the Auto-assign job

File: app/services/correspondence_auto_assigner.rb [hunk 3]
```diff
@@ -73,4 +78,8 @@ def assign_user_review_package_task(user:, task_params:)
   def permission_checker
     @permission_checker ||= OrganizationUserPermissionChecker.new
   end
+  
+   def correspondence_auto_assign_logger
+    @correspondence_auto_assign_logger ||= CorrespondenceAutoAssignLogger.new
```
Comment (zurbergram) [inline app/services/correspondence_auto_assigner.rb:83]: Pass in current user to initializer and update CorrespondenceAutoAssignLogger's initialize method to set user to an instance variable

**Review by zurbergram — CHANGES_REQUESTED** [review]
see comments

File: app/services/correspondence_auto_assign_logger.rb (hunk 1 above)
Comment (Kevma50287) [reply app/services/correspondence_auto_assign_logger.rb]: Added status parameter

File: app/services/correspondence_auto_assigner.rb (hunk 3 above)
Comment (Kevma50287) [reply app/services/correspondence_auto_assigner.rb]: revised to `@correspondence_auto_assign_logger ||= CorrespondenceAutoAssignLogger.new(@current_user)`

**Review by zurbergram — APPROVED** [review]

**Pull request merged**